### ISSUE TITLE
feat: hide copy text button on no JavaScript browsers

### ISSUE THREAD

**Issue by imkunet** [expressive-code/expressive-code#227]
## problem space

a keen viewer of your (hypothetical) site sees a code block with a copy button however finds that it is non-functional due to their hard line stance against JavaScript running in the browser.

## a potential solution

you write a (hypothetical) hacky `expressive-code` plugin to lazily patch this out as an issue but it's not enough (**don't use this since it will create a `noscript` block in every code block**, *or do but live a sub-optimal and complicit life*):

```ts
import { definePlugin } from 'astro-expressive-code';
import { h } from 'astro-expressive-code/hast';

// No JavaScript User
// hides the code copy button for those who don't use JavaScript.

export const noJavaScriptUser = () =>
  definePlugin({
    name: 'No JavaScript User',
    hooks: {
      postprocessRenderedBlockGroup: ({ renderData: { groupAst } }) => {
        groupAst.children.push(h('noscript', [h('style', 'figure.frame > div.copy { display: none; }')]));
      },
    },
  });
```

## the way out

the `expressive-code` `plugin-frames` plugin should build in support for hiding the copy button on `noscript` environments in a robust way that probably doesn't copy the `noscript` tag many times over multiple code blocks (like how it does in the above example).

## the astro solution

now I'm no `expressive-code` expert, but to fix this in https://github.com/withastro/astro it might be able to be done by just adapting the code earlier in this issue into somewhere above:

https://github.com/expressive-code/expressive-code/blob/84cdd77f3d6c25ef1836af6cf7420fe9053116b6/packages/astro-expressive-code/src/renderer.ts#L58

just convert from the https://github.com/syntax-tree/hastscript `h` call (from the code above) to using the object initializer!

obviously, this won't work on other targets.

## concerns

as pointed out by [a stackoverflow commenter](https://stackoverflow.com/questions/30319/is-there-a-html-opposite-to-noscript#comment1055061_431554) in the year 2009, this sort of *hack* is:
> Not allowed in the xhtml standard at least.

every browser I've tried this on seems fine with this.

**Comment by imkunet** [comment]
An alternate solution could lie in https://caniuse.com/mdn-css_at-rules_media_scripting. As of writing this comment, $85.64\\%$ of users can use this css rule to hide the copy button.

**Comment by hippotastic** [comment]
Hello and welcome!

I can imagine adding the CSS media query solution `@media (scripting: none) { ... }` to hide the copy button in case scripting is disabled. This approach seems to be a reasonably small change that would not require any additional HTML to be sent out for every code block.

Does the method you're using to disable JS get properly detected by [this example on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/scripting#result)?

**Comment by imkunet** [comment]
Interestingly enough, the MDN example linked doesn't seem to load with `javascript.enabled = false` on Firefox. I have given it a try though (on Firefox) and the media query seems to work just fine for detecting the `about:config` flag.

**Comment by hippotastic** [comment]
Fixed in #230. Thank you for the PR and welcome to the circle of Expressive Code contributors!
